### DISCUSSION TITLE
CFG Mimicking

### DISCUSSION THREAD

**Discussion by Bra2ha** [lllyasviel/Fooocus#3444] [Q&A] [answered]
_CFG Mimicking from TSNR_
_Enabling Fooocus's implementation of CFG mimicking for TSNR (effective when real CFG > mimicked CFG)._

Do I understand correctly that you want mimicked CFG to be lower than Guidance Scale?
If yes then why by default mimicked CFG is 7 and Guidance Scale is 4?

**Comment by mashb1t** [comment, marked as answer]
No, it's the opposite way, you want to use real CFG (guidance scale) until the threshold of meeting mimicked CFG, then if real CFG is above this you want to use mimicked CFG to not overshoot color saturation / clarity / etc.

example with mimicked CFG:
| guidance scale | uses mimicked CFG instead|
|--------|--------|
| 4 (< 7)| false|
| 7| false (same) |
| 7.1 (> 7) | true|
| 15 (> 7) | true| 

> why by default mimicked CFG is 7 and Guidance Scale is 4?

Default of 4 is below the mimicked CFG, so no mimicking takes place. It would if the real CFG would be > 7 though

**Reply by Bra2ha** [reply]
Understood, thank you

**Reply by IPv6** [reply]
would be cool to have CFG++ https://arxiv.org/abs/2406.08070 which maps CFG usage in (0,1) interval finally... as it should be logically from the very beginning, imho
